docs/docs/dev_docs/testing/cheat_codes.md

```diff
@@ -0,0 +1,458 @@
+
+// Dumps the current Ethereum chain state to a given file. 
+public async dumpChainState(fileName: string): Promise<void> 
+
+// Loads the Ethereum chain state from a file. You may use `dumpChainState()` to save the state of the Ethereum chain to a file and later load it. 
+public async loadChainState(fileName: string): Promise<void> 
+
+// Load the value at a storage slot of a contract address on Ethereum
+public async load(contract: EthAddress, slot: bigint): Promise<bigint> 
+
+// Set the value at a storage slot of a contract address on Ethereum (e.g. modify a storage variable on your portal contract or even the rollup contract). 
+public async store(contract: EthAddress, slot: bigint, value: bigint): Promise<void> 
+
+// Computes the slot value for a given map and key on Ethereum. A convenient wrapper to find the appropriate storage slot to load or overwrite the state.
+public keccak256(baseSlot: bigint, key: bigint): bigint
+
+// Let you send transactions on Ethereum impersonating an externally owned or contract, without knowing the private key.
+public async startImpersonating(who: EthAddress): Promise<void> 
+
+// Stop impersonating an account on Ethereum that you are currently impersonating.
+public async stopImpersonating(who: EthAddress): Promise<void> 
+
+// Set the bytecode for a Ethereum contract
+public async etch(contract: EthAddress, bytecode: `0x${string}`): Promise<void> 
+
+// Get the bytecode for a Ethereum contract
+public async getBytecode(contract: EthAddress): Promise<`0x${string}`> 
+```
+
+### blockNumber
+
+#### Function Signature 
+```js
+public async blockNumber(): Promise<number>
+```
+
+#### Description
+Fetches the current Ethereum block number.
+
+#### Example
+```js
+const blockNumber = await cc.eth.blockNumber()
+```
+
+### chainId
+
+#### Function Signature 
+```js
+public async chainId(): Promise<number>  
+```
+
+#### Description
+Fetches the Ethereum chain ID
+
+#### Example 
+```js
+const chainId = await cc.eth.chainId()
+```
+
+### timestamp
+
+#### Function Signature
+```js
+public async timestamp(): Promise<number> 
+```
+
+#### Description 
+Fetches the current Ethereum timestamp.
+
+#### Example
+```js 
+const timestamp = await cc.eth.timestamp()
+```
+
+### mine
+
+#### Function Signature
+```js
+public async mine(numberOfBlocks = 1): Promise<void> 
+```
+
+#### Description
+Mines the specified number of blocks on Ethereum (default 1).
+
+#### Example
+```js
+const blockNum = await cc.eth.blockNumber();
+await cc.eth.mine(10) // mines 10 blocks
+const newBlockNum = await cc.eth.blockNumber(); // = blockNum + 10.
+```
+
+### setNextBlockTimestamp
+
+#### Function Signature  
+```js
+public async setNextBlockTimestamp(timestamp: number): Promise<void>
+```
+
+#### Description
+Sets the timestamp (unix format in seconds) for the next mined block on Ethereum.
+Remember that timestamp can only be set in the future and not in the past.
```

Comment:
nit  its probably worth being explicit about the erroring

Reply:
Probably related to: https://github.com/AztecProtocol/aztec-packages/pull/1585#discussion_r1297072567